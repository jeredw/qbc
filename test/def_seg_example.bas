    DEF SEG = 0
    Status% = PEEK(&H417)              'Read keyboard status.
    POKE &H417, (Status% XOR &H40)     'Change Caps Lock state, bit 6.
