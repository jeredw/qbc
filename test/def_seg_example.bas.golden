(program 
  (statement 
    (def_seg_statement DEF SEG = 
      (expr 
        (literal 0)))) \n 
  (statement 
    (assignment_statement 
      (variable_or_function_call Status%) = 
      (expr 
        (variable_or_function_call PEEK 
          (args_or_indices 
            ( 
              (expr 
                (literal &H417)) )))))) \n 
  (statement 
    (call_statement 
      (untyped_id POKE) 
      (call_argument_list 
        (call_argument 
          (expr 
            (literal &H417))) , 
        (call_argument 
          (expr 
            ( 
              (expr 
                (expr 
                  (variable_or_function_call Status%)) XOR 
                (expr 
                  (literal &H40))) )))))) \n)
